packages/fscomponents/src/components/Form/Templates/above/LabelAboveStatefulTextbox.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react';
+import { Image, Text, TextInput, View } from 'react-native';
+import {
+  defaultTextboxStyle,
+  errorIcon,
+  getColor,
+  StatefulTextboxProps,
+  StatefulTextboxState,
+  successIcon
+} from '../formBoilerplate';
+
+export default class LabelAboveTextbox extends Component<StatefulTextboxProps,
+StatefulTextboxState> {
+
+  state: StatefulTextboxState = {
+    active: false,
+    validated: false
+  };
+
+  onFocus = () => {
+    this.setState({
+      active: true,
+      validated: false
+    });
+  }
+
+  onBlur = () => {
+    this.setState({
+      active: false,
+      validated: true
+    });
+  }
+
+  render(): JSX.Element {
+    const {locals} = this.props;
+
+    const prevonBlur = locals.onBlur;
+    const prevOnFocus = locals.onFocus;
+
+    locals.onBlur = () => {
```

Comment:
Can we move this into `getDerivedStateFromProps`?

Reply:
I discussed this with Sarah and, after some reading, it looks like this may be a good candidate for memoize: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization

She's going to take a stab at doing this then we can revisit.